docker/container.go

```diff
@@ -96,6 +97,51 @@ func name(names []string) string {
 	return current[1:]
 }
 
+// Rebuild will delete the container and create a new one with the current configuration
+func (c *Container) Rebuild(imageName string) (*dockerclient.APIContainers, error) {
+	info, err := c.findInfo()
+	if err != nil {
+		return nil, err
+	} else if info == nil {
+		return nil, fmt.Errorf("Can not find container to rebuild for service: %s", c.service.Name())
+	}
+
+	hash := info.Config.Labels[HASH.Str()]
+	if hash == "" {
+		return nil, fmt.Errorf("Failed to find hash on old container: %s", info.Name)
+	}
+
+	name := info.Name[1:]
+	newName := fmt.Sprintf("%s_%s", name, info.ID[:12])
+	deleted := false
+	logrus.Debugf("Renaming %s => %s", name, newName)
+	if err := c.client.RenameContainer(dockerclient.RenameContainerOptions{ID: info.ID, Name: newName}); err != nil {
+		logrus.Errorf("Rename failed, deleting %s", name)
+		if err := c.client.RemoveContainer(
+			dockerclient.RemoveContainerOptions{ID: info.ID, Force: true, RemoveVolumes: false}); err != nil {
+			return nil, err
```

Comment:
This is wrong. If rename fails you can't remove the old container. That would cause you to lose your volumes (unless you go in and manually copy them over somewhere, which wasn't possible before 1.9).